pages/form.tsx

```diff
@@ -0,0 +1,114 @@
+              );
+            } catch (e) {
+              console.log(`onSubmit(): ${e}`);
+            }
+          }}
+        >
+          <div>
+            <h1 className="font-bold text-4xl mt-20 dark:text-ocf-gray">
+              Panel Details
+            </h1>
+          </div>
+          <div className="my-8">
+            <h2 className="font-medium text-lg dark:text-ocf-gray">
+              Solar panel direction
+            </h2>
+            <p className="text-ocf-gray-900 text-xs mt-1 dark:text-ocf-gray-800">
+              (0º = North, 90º = East, 180º = South, 270º = West)
+            </p>
+            <input
+              required
+              name="solarPanelDirection"
+              type="number"
+              placeholder="170º"
+              className="mt-3 text-lg text-center 
+              bg-gray-50 border border-gray-300 hover:ring-ocf-yellow hover:border-ocf-yellow  text-gray-900 rounded-lg  block w-20 h-14 p-2.5               
+              dark:bg-ocf-gray-800
+              dark:border-ocf-gray-800
+              dark:placeholder-ocf-gray-900 dark:text-ocf-gray"
+            />
+            <p className="text-ocf-gray-900 text-xs mt-1 underline dark:text-ocf-gray-800">
+              I don&apos;t know
+            </p>
+          </div>
+
+          <div className="my-8">
+            <h2 className="font-medium text-lg dark:text-ocf-gray">
+              Solar panel angle tilt
+            </h2>
+            <p className="text-ocf-gray-900 text-xs mt-1 dark:text-ocf-gray-800">
+              (Degrees above the horizontal)
+            </p>
+            <input
+              required
+              name="solarPanelAngelTilt"
+              type="number"
+              placeholder="30º"
+              className="mt-3 text-lg text-center 
+              bg-gray-50 border border-gray-300 text-gray-900 rounded-lg block w-20 h-14 p-2.5 hover:ring-ocf-yellow hover:border-ocf-yellow  dark:bg-ocf-gray-800
+              dark:border-ocf-gray-800
+              dark:placeholder-ocf-gray-900 dark:text-ocf-gray "
+            />
+          </div>
+
+          <div className="my-8">
+            <div className="flex flex-row justify-start items-center w-full">
+              <h2 className="font-medium text-lg dark:text-ocf-gray">
+                Solar panel capacity
+              </h2>
+              <p className="text-ocf-gray-900 text-xs mx-1 dark:text-ocf-gray-800">
+                (optional)
+              </p>
+            </div>
+            <input
+              name="solarPanelCapacity"
+              type="number"
+              placeholder="3000 W"
+              className="mt-3 text-lg text-center bg-gray-50 border border-gray-300 text-gray-900 rounded-lg hover:ring-ocf-yellow hover:border-ocf-yellow block w-28 h-14 p-2.5 dark:bg-ocf-gray-800
+              dark:border-ocf-gray-800
+              dark:placeholder-ocf-gray-900 dark:text-ocf-gray"
+            />
+          </div>
```

Comment:
I'd love to get some extra validation on the angle and capacity input elements—seems like even though the API will take in any reals, these logically should only be positive floats. Adding 
```min="0" onkeypress="return ((event.charCode >= 48 && event.charCode <= 57) || event.charCode === 46)"```
should do the trick here.

Reply:
This is a good catch, we could even restrict the angle input with a `max=359`. I think the `type=number` attribute should be enough instead of the `onkeypress`.

Reply:
I think the onkeypress is necessary here unfortunately-with just min=0 you're unable to use the arrows in the input to go below zero, but negatives can be typed through keyboard input just fine. Adding the onkeypress basically forbids the '-' sign from being entered.

If there's a more elegant way to do this that'd be great, but this is the only way I've been able to get our expected behavior for now.

Reply:
That's true... on mobile chrome for me at least the virtual keyboard doesn't even show a `-` for some reason. Also, if we apply a red border color for "[invalid](https://tailwindcss.com/docs/hover-focus-and-other-states#form-states)" (value < min or > max) it could indicate the issue (or a max issue). Also, `min` will prevent submission if the value they input is lower than it. We can use `invalid` in combination with [placeholder-shown](https://tailwindcss.com/docs/hover-focus-and-other-states#form-states) to show the red border only if they have something input (rather than a currently blank required field).

That all might be overkill though. I'm fine with the keypress solution if we can turn those keycode numbers into constants or replace with them with their string representation.

Reply:
`onKeyPress` is now [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Element/keypress_event). Nevertheless, there are other ways to restrict input to positive floats between 0 and 359 (inclusive).